scripts/update_wallet_detector.js

```diff
@@ -0,0 +1,82 @@
+// Script to add a new wallet version (code and/or implementation) to the ArgentWalletDetector.
+//
+// ////////////////////////// WARNING ///////////////////////////////////////////////
+// There is only one instance deployed that detects wallets for both staging and prod.
+// //////////////////////////////////////////////////////////////////////////////////
+//
+// To add a new wallet (code + implementation):
+// ./execute_script.sh update_wallet_detector.js <network> --wallet <wallet address>
+//
+// To add a new implementation:
+// ./execute_script.sh update_module_registry.js <network> --implementation <wallet implementation>
+//
+// To add a new code:
+// ./execute_script.sh update_module_registry.js <network> --code <wallet code>
+//
+// where:
+//    - network = [test, prod]
+// ////////////////////////////////////////////////////////////////////
+
+const ArgentWalletDetector = require("../build/ArgentWalletDetector");
+const MultiSig = require("../build/MultiSigWallet");
+
+const DeployManager = require("../utils/deploy-manager.js");
+const MultisigExecutor = require("../utils/multisigexecutor.js");
+
+async function main() {
+  let wallet;
+  let code;
+  let implementation;
+
+  // Read Command Line Arguments
+  let idx = process.argv.indexOf("--network");
+  const network = process.argv[idx + 1];
+
+  idx = process.argv.indexOf("--wallet");
+  if (idx > 0) {
+    wallet = process.argv[idx + 1];
+  } else {
+    idx = process.argv.indexOf("--implementation");
+    if (idx > 0) {
+      implementation = process.argv[idx + 1];
+    } else {
+      idx = process.argv.indexOf("--code");
+      if (idx > 0) {
+        code = process.argv[idx + 1];
+      } else {
+        console.log("Error: No argument provided");
+        return;
+      }
+    }
+  }
+
+  // Setup deployer
+  const manager = new DeployManager(network);
+  await manager.setup();
+  const { configurator } = manager;
+  const { deployer } = manager;
+  const deploymentWallet = deployer.signer;
+  const { config } = configurator;
+
+  const ArgentWalletDetectorWrapper = await deployer.wrapDeployedContract(ArgentWalletDetector, config.contracts.ArgentWalletDetector);
+  const MultiSigWrapper = await deployer.wrapDeployedContract(MultiSig, config.contracts.MultiSigWallet);
+  const multisigExecutor = new MultisigExecutor(MultiSigWrapper, deploymentWallet, config.multisig.autosign);
+
+  if (wallet) {
+    console.log(`Adding wallet code and implementation from ${wallet}`);
+    await multisigExecutor.executeCall(ArgentWalletDetectorWrapper, "addCodeAndImplementationFromWallet", [wallet]);
+  }
+  if (code) {
+    console.log(`Adding wallet code ${code}`);
+    await multisigExecutor.executeCall(ArgentWalletDetectorWrapper, "addCode", [code]);
+  }
+  if (implementation) {
+    console.log(`Adding wallet implementation ${implementation}`);
+    await multisigExecutor.executeCall(ArgentWalletDetectorWrapper, "addImplementation", [implementation]);
+  }
```

Comment:
Can we switch this if logic to be 
```
if (wallet) { 
...
} else if (code) { 
... 
} else if (implementation) {
...
}
```
to indicate we can have 1 of 3 option only as opposed to any combination of them. Also do the same for the nested `if`s  [above](https://github.com/argentlabs/argent-contracts/pull/156/files?file-filters%5B%5D=.js&file-filters%5B%5D=.sol#diff-745dc33a43b9e53984f71fe20dc8aa40435bab5d8fa4d737a41253491234249eR36) in 

```
if (idx > 0) {
    wallet = process.argv[idx + 1];
  } else {
    idx = process.argv.indexOf("--implementation");
    if (idx > 0) {
...
```

Reply:
Agreed and done for the first part of your comment, but the second part is already in the form of an `if else`.